File: LeetCode/easy/102.BinaryTreeLevelOrderTraversal/BinaryTreeLevelOrderTraversal.java
/*Betas 88.5%
*/
public class BinaryTreeLevelOrderTraversal{
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> list = new ArrayList<List<Integer>>();
        traversalByLevel(root,list,1);
        return list;
    }
    public void traversalByLevel(TreeNode root,List<List<Integer>> list,int level){
        if(root==null) return;
        if(list.size()<level) list.add(new ArrayList<Integer>());
        list.get(level-1).add(root.val);
        traversalByLevel(root.left,list,level+1);
        traversalByLevel(root.right,list,level+1);
    }
}

class TreeNode {
    int val;
    TreeNode left;
    TreeNode right;
    TreeNode(int x) { val = x; }
}

File: LeetCode/medium/102. Binary Tree Level Order Traversal/BinaryTreeLevelOrderTraversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
class Solution {
   public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        if (root == NULL) {
            return result;
        }
        queue<TreeNode*> q;
        q.push(root);
        while (!q.empty()) {
            vector<int> level;
            int len = q.size();
            while (len--) {
                TreeNode* node = q.front();
                q.pop();
                level.push_back(node->val);
                if (node->left != NULL) {
                    q.push(node->left);
                }
                if (node->right != NULL) {
                    q.push(node->right);
                }
            }
            result.push_back(level);
        }
        return result;
    }
};